src/odemis/gui/cont/tabs.py

```diff
@@ -2034,8 +2034,68 @@ def display_new_data(self, filename, data, extend=False):
                                 self.tab_data_model.ar_cal.value)
                 self.tab_data_model.ar_cal.value = u""  # remove the calibration
 
-        # TODO: if all the views are either empty or contain the same streams,
+        # if all the views are either empty or contain the same streams,
         # display in full screen by default (with the first view which has streams)
+        # Show the 1 x 1 view if:
+        # - the streams are all the same in every view 
+        # - views are empty with no projection
+        
+        def compare_projection_lists(list1, list2):
+            """ Returns true if two projection lists are the same, 
+            i.e. they are projecting the same streams, even if the objects are different """
+            if len(list1) != len(list2):
+                return False
+            
+            matches = 0
+            # since it is impossible to order the lists, use o(n^2) comparison
+            # each proj in list1 should have at least one match in list2
+            for proj1 in list1:
+                match = False
+                # there should be at least one corresponding match in the second list
+                for proj2 in list2:
+                    # matches are of the same type projecting the same stream reference
+                    if type(proj1) == type(proj2) and proj1.stream == proj2.stream:
+                        match = True
+                        break
+                    
+                if not match:
+                    return False
+                else:
+                    matches += 1
+                
+            assert(matches == len(list1))
+            return True
+
+        same_projections = []
+        same_views = 0
+        empty_views = 0
+        first_non_empty_view = None
+        for view in self.tab_data_model.visible_views.value:
+            view_projections = view.getProjections()
+
+            # if no projections, the view is empty
+            if not view_projections:
+                empty_views += 1
+                continue
+            else:
+                # if this list is empty, we are at the first non empty view
+                if not same_projections:
+                    same_projections = view_projections
+                    same_views += 1
+                    first_non_empty_view = view
+                else:
+                    # if there is any difference, then there are different projections and we can abort early
+                    if compare_projection_lists(same_projections, view_projections):
+                        same_views += 1
+                        continue
+                    else:
+                        break
```

Comment:
Did you check what happens if you have two streams are duplicated twice?

Reply:
I think it will just focus on the first, which is ok and gives the intended effect.